pkg/controller/infra/infra_controller.go

```diff
@@ -118,99 +138,369 @@ type InfraController struct {
+	}
+	if cluster == nil {
+		return nil
+	}
+
+	// only proceed if this is the current provisioning job for the cluster
+	if cluster.Status.ProvisioningJob == nil || cluster.Status.ProvisioningJob.Name != job.Name {
+		return nil
+	}
+
+	return c.syncClusterStatusWithJob(cluster, job)
+}
+
+func (c *InfraController) syncClusterStatusWithJob(cluster *clusteroperator.Cluster, job *v1batch.Job) error {
+	cluster = cluster.DeepCopy()
+	now := metav1.Now()
+	jobCompleted := jobCondition(job, v1batch.JobComplete)
+	if jobCompleted != nil {
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		if clusterProvisioning != nil &&
+			clusterProvisioning.Status == kapi.ConditionTrue {
+			clusterProvisioning.Status = kapi.ConditionFalse
+			clusterProvisioning.LastTransitionTime = now
+			clusterProvisioning.LastProbeTime = now
+			clusterProvisioning.Reason = "JobCompleted"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime)
+		}
+		clusterProvisioned := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioned)
+		if clusterProvisioned != nil &&
+			clusterProvisioned.Status == kapi.ConditionFalse {
+			clusterProvisioned.Status = kapi.ConditionTrue
+			clusterProvisioned.LastTransitionTime = now
+			clusterProvisioned.LastProbeTime = now
+			clusterProvisioned.Reason = "JobCompleted"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime)
+			cluster.Status.Provisioned = true
+		}
+		if clusterProvisioned == nil {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, clusteroperator.ClusterCondition{
+				Type:               clusteroperator.ClusterHardwareProvisioned,
+				Status:             kapi.ConditionTrue,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             "JobCompleted",
+				Message:            fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime),
+			})
+			cluster.Status.Provisioned = true
+		}
+		cluster.Status.ProvisioningJob = nil
+	}
+	jobFailed := jobCondition(job, v1batch.JobFailed)
+	if jobFailed != nil {
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		if clusterProvisioning != nil &&
+			clusterProvisioning.Status == kapi.ConditionTrue {
+			clusterProvisioning.Status = kapi.ConditionFalse
+			clusterProvisioning.LastTransitionTime = now
+			clusterProvisioning.LastProbeTime = now
+			clusterProvisioning.Reason = "JobFailed"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s failed at %v, reason: %s", job.Namespace, job.Name, jobFailed.LastTransitionTime, jobFailed.Reason)
+		}
+		cluster.Status.ProvisioningJob = nil
+	}
+
+	if jobCompleted == nil && jobFailed == nil && job.Status.Active > 0 {
```

Comment:
I'm not sure about the check for `job.Status.Active > 0` here? If we are going to change the reason for the provisioning condition to JobRunning based upon there being active pods for the job, then we should also change it to something like JobWaiting when there are no active pods for the job. I am not sure whether any check for the number of active pods is really necessary, though, and whether it might be sufficient to just set the reason to JobRunning so long as the job is not completed or failed.

Reply:
yeah, I think if it's not completed or failed, it may be safe to assume that is still active